File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n =nums.length;
        int low = 0, mid =0, high=n-1;

        while(mid<=high){
            if(nums[mid]==0){
                swap(nums, mid, low);
                low++;
                mid++;
            }
            else if(nums[mid]==1){
                mid++;
            }
            else{       //mid==2
                swap(nums, high, mid);
                high--;
            }
        }
    }    
    private void swap(int[] nums , int small, int big){
        int temp = nums[small];
        nums[small] = nums[big];
        nums[big] = temp;
    }
}



//  if (n==0 || n==1){
//             return;
//         } 
//         for(int i = 0 ; i<n-1 ; i++){
//             for(int j = 0 ; j<n-1-i ;j++){
//                 if (nums[j]>nums[j+1]){
//                     temp = nums[j+1];
//                     nums[j+1]=nums[j];
//                     nums[j]=temp;
//                 }
//             }
//         }

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        n=len(nums)
        low=0
        mid=0
        high = n-1
        while mid<=high:
            if nums[mid]==0:
                temp = nums[low]
                nums[low]=nums[mid]
                nums[mid]=temp
                low+=1
                mid+=1
            elif nums[mid]==1:
                mid+=1
            else:           #nums[mid]==2
                temp = nums[high]
                nums[high]=nums[mid]
                nums[mid]=temp
                high-=1
